doc/resource_types.md

```diff
@@ -1613,6 +1613,21 @@ end
 ```
 
 
+### have_lifecycle_rule
```

Comment:
Append doc to `doc/_resource_types/s3_bucket.md` and run `bundle exec bin/toolbox docgen > doc/resource_types.md` 
( see https://github.com/k1LoW/awspec/blob/master/doc/contributing.md )